src/lib/edge-functions/proxy.js

```diff
@@ -74,16 +78,27 @@ const initializeProxy = async ({
     projectDir,
   })
   const hasEdgeFunctions = userFunctionsPath !== undefined || internalFunctions.length !== 0
+  let hasServerError = false
 
   return async (req) => {
-    if (req.headers[headers.Passthrough] !== undefined || !hasEdgeFunctions) {
+    if (req.headers[headers.Passthrough] !== undefined || !hasEdgeFunctions || hasServerError) {
       return
     }
 
-    const [geoLocation, { registry }] = await Promise.all([
-      getGeoLocation({ mode: geolocationMode, offline, state }),
-      server,
-    ])
+    let promiseResult
+
+    try {
+      promiseResult = await Promise.all([getGeoLocation({ mode: geolocationMode, offline, state }), server])
+    } catch (error_) {
+      error(error_ instanceof Error ? error_ : format(error_), { exit: false })
+      hasServerError = true
+    }
+
+    if (promiseResult === undefined) {
```

Comment:
This only happens if we fall into the `catch` above, right? Can we move it there? I think it would be a bit easier to read.